File: app/src/main/java/piuk/blockchain/android/ui/send/SendView.java
package piuk.blockchain.android.ui.send;

import android.os.Bundle;
import android.support.annotation.ColorRes;
import android.support.annotation.Nullable;
import android.support.annotation.StringRes;

import piuk.blockchain.android.data.contacts.models.PaymentRequestType;
import piuk.blockchain.android.ui.account.PaymentConfirmationDetails;
import piuk.blockchain.android.ui.base.View;
import piuk.blockchain.android.ui.customviews.ToastCustom;

interface SendView extends View {

    Bundle getFragmentBundle();

    @Nullable
    String getClipboardContents();

    void hideSendingAddressField();

    void hideReceivingAddressField();

    void updateBtcAmount(String amount);

    void updateBtcUnit(String unit);

    void updateFiatUnit(String unit);

    void onSetSpendAllAmount(String textFromSatoshis);

    void showInvalidAmount();

    void onShowSpendFromWatchOnly(String address);

    void updateBtcTextField(String text);

    void updateFiatTextField(String text);

    void onShowPaymentDetails(PaymentConfirmationDetails confirmationDetails, @Nullable String note);

    void onShowReceiveToWatchOnlyWarning(String address);

    void showToast(@StringRes int message, @ToastCustom.ToastType String toastType);

    void onShowTransactionSuccess(@Nullable String mdid,
                                  @Nullable String fctxId,
                                  String hash,
                                  long transactionValue);

    void showBroadcastFailedDialog(String mdid, String txHash, String facilitatedTxId, long transactionValue);

    void showBroadcastSuccessDialog();

    void onShowBIP38PassphrasePrompt(String scanData);

    void finishPage(boolean paymentToContactMade);

    void setContactName(String name);

    void showProgressDialog(@StringRes int title);

    void dismissProgressDialog();

    void setDestinationAddress(String btcAddress);

    void setMaxAvailable(String max);

    void setMaxAvailableColor(@ColorRes int color);

    void setMaxAvailableVisible(boolean visible);

    void setUnconfirmedFunds(String notice);

    void showSecondPasswordDialog();

    void navigateToAddNote(PaymentConfirmationDetails paymentConfirmationDetails,
                           PaymentRequestType paymentRequestType,
                           String contactId,
                           long satoshis,
                           int accountPosition);

    void dismissConfirmationDialog();

    void updateFeeField(String fee);

    long getCustomFeeValue();

    void lockContactsFields();
}

File: app/src/main/java/piuk/blockchain/android/ui/send/SendView.kt
package piuk.blockchain.android.ui.send

import android.support.annotation.ColorRes
import android.support.annotation.Nullable
import android.support.annotation.StringRes
import piuk.blockchain.android.data.currency.CryptoCurrencies
import piuk.blockchain.android.ui.account.PaymentConfirmationDetails
import piuk.blockchain.android.ui.base.View
import java.util.*

interface SendView : View {

    val locale: Locale

    //Update field
    fun updateSendingAddress(label: String)

    fun updateReceivingHint(hint: Int)

    fun updateCryptoCurrency(currency: String)

    fun updateFiatCurrency(currency: String)

    fun updateCryptoAmount(amountString: String?)

    fun updateFiatAmount(amountString: String?)

    fun updateWarning(message: String)

    fun updateMaxAvailable(maxAmount: String)

    fun updateMaxAvailableColor(@ColorRes color: Int)

    fun updateReceivingAddress(address: String)

    fun updateFeeAmount(fee: String)

    //Set property
    fun setCryptoMaxLength(length: Int)

    fun setTabSelection(tabIndex: Int)

    fun setFeePrioritySelection(index: Int)

    fun clearWarning()

    //Hide / Show
    fun hideReceivingDropdown()

    fun showReceivingDropdown()

    fun hideSendingFieldDropdown()

    fun showSendingFieldDropdown()

    fun showMaxAvailable()

    fun hideMaxAvailable()

    fun showFeePriority()

    fun hideFeePriority()

    //Enable / Disable
    fun disableCryptoTextChangeListener()

    fun enableCryptoTextChangeListener()

    fun disableFiatTextChangeListener()

    fun enableFiatTextChangeListener()

    fun enableFeeDropdown()

    fun disableFeeDropdown()

    fun setSendButtonEnabled(enabled: Boolean)

    fun disableInput()

    fun enableInput()

    // Fetch value
    fun getCustomFeeValue(): Long

    fun getClipboardContents(): String?

    fun getReceivingAddress(): String?

    fun getFeePriority(): Int

    // Prompts
    fun showSnackbar(@StringRes message: Int, duration: Int)

    fun showSnackbar(message: String, @Nullable extraInfo: String?, duration: Int)

    fun showEthContractSnackbar()

    fun showBIP38PassphrasePrompt(scanData: String)

    fun showWatchOnlyWarning(address: String)

    fun showProgressDialog(@StringRes title: Int)

    fun showSpendFromWatchOnlyWarning(address: String)

    fun showSecondPasswordDialog()

    fun showPaymentDetails(confirmationDetails: PaymentConfirmationDetails, note: String?, allowFeeChange: Boolean)

    fun showLargeTransactionWarning()

    fun showTransactionSuccess(hash: String, transactionValue: Long, cryptoCurrency: CryptoCurrencies)

    fun dismissProgressDialog()

    fun dismissConfirmationDialog()

    fun finishPage()
}
